packages/bsky/src/services/indexing/plugins/post.ts

```diff
@@ -243,6 +244,7 @@ const notifsForInsert = (obj: IndexedPost) => {
   }
 
   for (const ancestor of obj.ancestors ?? []) {
+    if (obj.post.violatesThreadGate) continue // don't notify thread when post violates threadgate
```

Comment:
this will always do a `continue` for every ancestor - not a big deal, but we could put the check before both for loops & just return earlier